aspnetcore/mvc/controllers/routing.md

```diff
@@ -70,92 +61,169 @@ public class HomeController : Controller
 }
 ```
 
-Using this controller definition and route template, the `HomeController.Index` action would be executed for any of the following URL paths:
+Using the preceding controller definition and route template, the `HomeController.Index` action is run for the following URL paths:
 
 * `/Home/Index/17`
-
 * `/Home/Index`
-
 * `/Home`
-
 * `/`
 
-The convenience method `UseMvcWithDefaultRoute`:
+The URL path `/` uses the route template default `Home` controllers and `Index` action. The URL path `/Home` uses the route template default `Index` action.
+
+The convenience method <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapDefaultControllerRoute*>:
 
 ```csharp
 endpoints.MapDefaultControllerRoute();
 ```
 
-Can be used to replace:
+Replaces:
 
 ```csharp
 endpoints.MapControllerRoute("default", "{controller=Home}/{action=Index}/{id?}");
 ```
 
-Routing is configured using the `UseRouting()` and `UseEndpoints()` middleware. To use controllers:
-* Call `MapControllers()` inside `UseEndpoints()` to map attribute-routed controllers.
-* Call `MapControllerRoute()` (or similar) to map conventionally-routed controllers.
+Routing is configured using the <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting*> and <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints*> middleware. To use controllers:
 
-Using `MapControllerRoute()` (or similar) will also map attribute-routed controllers.
+* Call <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllers*> inside `UseEndpoints` to map [attribute routed](#ar) controllers.
+* Call <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllerRoute*> or <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapAreaControllerRoute*>, to map [conventionally routed](#cr) controllers.
```

Comment:
We should add a callout here to the effect of:

MVC's routing integration has two different programming experiences, conventional routing and attribute routing. There is one common routing implementation that is used regardless of the way in which it's programmed. Conventional routing concepts and attribute routing concepts are used to program the underlying routing system.

Reply:
Nevermind. This is already here later in the doc.